ogr/ogrsf_frmts/generic/ogrlayer.cpp

```diff
@@ -227,6 +227,85 @@ OGRErr OGRLayer::GetExtent(int iGeomField, OGREnvelope *psExtent, int bForce)
         return GetExtentInternal(iGeomField, psExtent, bForce);
 }
 
+OGRErr OGRLayer::GetExtent3D(int iGeomField, OGREnvelope3D *psExtent3D,
+                             int bForce)
+
+{
+    psExtent3D->MinX = 0.0;
+    psExtent3D->MaxX = 0.0;
+    psExtent3D->MinY = 0.0;
+    psExtent3D->MaxY = 0.0;
+    psExtent3D->MinZ = 0.0;
+    psExtent3D->MaxZ = 0.0;
```

Comment:
we might want to initialize those to NaN in case there would be only 2D geometries
```suggestion
    psExtent3D->MinZ = 0.0;
    psExtent3D->MaxZ = 0.0;
```

Reply:
Ok, I'll implement it to handle NaN for Z quietly. 
Client code will need to check for Z not NaN to make sure it's 3D.